rvic/convert.py

```diff
@@ -39,6 +39,9 @@ def convert(config_file):
         uhs2param_final(outlets, dom_data, new_dom_data, config_dict, directories)
         # ---------------------------------------------------------------- #
 
+    except BaseException as e:
```

Comment:
Does this do anything? Logging the Exception and then raising it, is basically the same thing as letting it propagate up, no?

Reply:
I'm gonna have a closer look at what he was working on.

Reply:
So, he added these `try`/`except`/`finally` blocks in order to [ensure the logger is closed even in case any of the process terminate prematurely](https://github.com/pacificclimate/rvic-daccs/issues/6)

Reply:
Yeah, the `try`/`finally` block, is absolutely worthwhile. You want to make sure that resources get closed gracefully if there's an exception raised in the `try` block. My question/contention is: is there any value to the `except` block? That block simply catches the exception, logs it and raises it. But (I *think*) it gets logged or at least output after you raise it as well. Execute these two simple examples to compare the behvaviour and see if there's any value added for the `except` block:

```python3
try:
    print("try")
    raise(Exception("I've been raised!!!"))
finally:
    print("This should *aways* execute.")
```

```python3
try:
    print("try")
    raise(Exception("I've been raised!!!"))
except Exception as e:
    raise(e)
finally:
    print("This should *aways* execute.")
```

Reply:
Without the `except` block, the exception doesn't get logged to the RVIC logger.

Reply:
OK, thanks for clarifying. That's what I needed to know :)

Reply:
No prob. Good way for me to get a better idea of what Sangwon did. So, should I leave this as it is?

Reply:
Oh, you resolved it.